src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -324,7 +324,17 @@ CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::StoreRegulatoryLocation
 template <class ConfigClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetCountryCode(char * buf, size_t bufSize, size_t & codeLen)
 {
-    return ReadConfigValueStr(ConfigClass::kConfigKey_CountryCode, buf, bufSize, codeLen);
+    VerifyOrReturnError(bufSize > 2, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    CHIP_ERROR err = ReadConfigValueStr(ConfigClass::kConfigKey_CountryCode, buf, bufSize, codeLen);
+    if (err != CHIP_NO_ERROR || codeLen == 0)
+    {
+        strncpy(buf, "XX", bufSize);
+        buf[2]  = '\0';
```

Comment:
Don't need this, given bufSize > 2 and use of strncpy, right?

But better yet would be to use CopyString, which always guarantees null-termination.

Reply:
This change is not needed, since the real problem is the test case dose not reach to this API but to Attribute store directly. Need to intercept read and write API for location attribute.